### DISCUSSION TITLE
**Windows 11 Truffle Installation Guide**

### DISCUSSION THREAD

**Discussion by apdan510** [ConsenSys-archive/truffle#5089] [Installation]
### Truffle Installation guide for Windows 11 22H2

Last tested: November 17th, 2022

**Requirements:** 
- Windows 11 (22H2) 
- Windows 10 (22H2) 64-bit
- Node.js (NPM) v14 - v18
- Visual Studio C++ Build Tools 2017 or higher
- Python 2.7 or higher

**Step 1. Install Node.js Version 18**

Install Node.js with the official Windows installation MSI file from Nodejs.org: 
https://nodejs.org/dist/v18.12.1/node-v18.12.1-x64.msi

This complete .msi installer will install Node.js (NPM) and add the NPM command to your system's PATH. This .msi will also install dependencies and compilers needed to install Truffle:
Python Build tools, Visual Studio C++ build tools, etc.

![image](https://user-images.githubusercontent.com/40121410/202574521-893d0165-8187-4574-a4bb-57fd5053f161.png)
![image](https://user-images.githubusercontent.com/40121410/202574698-fa91efc2-ac4c-444b-91e9-78695ef8cf8c.png)
Click "next" on each prompt and leave all settings to its default install configuration

**Step 2. Install Python and Windows Build tools**
The second half of the Node.js .msi installer package comes with the dependencies and compiling tools necessary to install Truffle. When you reach this page please make sure the option is "checked" to install these tools. 

![image](https://user-images.githubusercontent.com/40121410/167053708-90980b60-2d94-4b94-b41b-73ca8fae2f30.png)
![image](https://user-images.githubusercontent.com/40121410/167054030-8d44e66e-7adb-4c80-a6e2-271cd5098522.png)
After the .msi installer has completed, a new window will pop-up to install python and the build tools.

![image](https://user-images.githubusercontent.com/40121410/167054352-9a96b844-a893-45ae-b1ee-b94b39ba6c91.png)
![image](https://user-images.githubusercontent.com/40121410/202576360-c373cb7b-41cb-48b8-8c57-e2ad8f2b21de.png)
![image](https://user-images.githubusercontent.com/40121410/202576759-17e6cb18-ae62-45d8-bb28-6909c3ef5b1c.png)
type "enter" into your terminal to complete installation

**Step 3. Install Truffle using NPM with the -g flag in your terminal**
To verify if Node.js is installed correctly, open a Powershell terminal with administrative privileges. You can do this by right clicking the Windows Start button and selecting "Windows Powershell (Admin)" 
![image](https://user-images.githubusercontent.com/40121410/167054926-5f0bfb49-3173-4943-ba16-2d06996f84ae.png)

Enter these commands into your Powershell Terminal: `node -v` and `npm -v` these commands should return the Node.js and NPM version if installed correctly. 
![image](https://user-images.githubusercontent.com/40121410/202577209-3d255535-004c-4a1c-a1a5-19d74fe28000.png)

Next, lets install the latest version of Truffle on your computer. We will use -g flag to install Truffle globally on your computer. This means that you can use the `truffle` command inside a terminal from any location or directory. 

To install Truffle enter this command in your Powershell terminal: `npm install -g truffle `
![image](https://user-images.githubusercontent.com/40121410/202577556-ad2ea4a9-0f3d-4988-877d-4ae5fd979c1a.png)

 During installation you may come across various warnings, these are typical and everything should run fine as long as there are no error messages.
![image](https://user-images.githubusercontent.com/40121410/202578033-61c392b3-c2d6-48e5-a72c-9e54fc17386c.png)

To verify to see if the Truffle installation is successful, enter this command in your Powershell terminal:  `truffle version` 
The terminal should return the installed Truffle version as well as Node.js information if the installation was successful. 
![image](https://user-images.githubusercontent.com/40121410/202578145-3eee9453-3d36-4123-91e1-1030f2121fd5.png)

**Comment by chidubemOkafor** [comment]
![Screenshot 2022-05-21 122406](https://user-images.githubusercontent.com/93622341/169670524-865150cc-434e-4535-9d39-69471fde4aef.png)
 how do I
 fix this?

**Reply by swisstackle** [reply]
Have you installed the latest version of visual studio?

**Reply by apdan510** [reply]
**This error is saying that there is no visual studio compiler is installed on your computer, please follow these steps to install it:**

**Open an administrator CMD line or Powershell**
**Make sure Chocolatey is installed by pasting the below command into an administrative Cmdline or Powershell:** 
`Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))`

**Install Visual studio 2019 build tools**
`choco install -y visualstudio2019buildtools visualstudio2019-workload-vctools`

**Set VS studio 2019 build tools**
`npm config set msvs_version 2019`

**Then try installing Truffle once more:**
`npm install -g truffle`

**Comment by Midsty** [comment]
![image](https://user-images.githubusercontent.com/61846381/171505178-353ab14f-935a-4beb-b365-66d996950c7f.png)

Am still getting the Visual Studio error after running the commands 

I already got Chocolatey and visual studio

**Reply by apdan510** [reply]
try this command to set the msvs version to your "VS build tools 2022" installed : `npm config set msvs_version 2022`

then try installing truffle again: `npm install -g truffle`

Hope this helps!

**Reply by SteffQing** [reply]
formerly, I was getting around 5 errors
Now I got a whole catalog of them

**Reply by apdan510** [reply]
Steve, can you please share those errors here please? Screenshots, errors, or logs. 

Kind regards,
Alex

**Comment by Midsty** [comment]
It kinda solved the problem but it can't see the truffle version:

![image](https://user-images.githubusercontent.com/61846381/171572651-df8e357d-aee1-4e84-bfb4-4f4409db4880.png)

**Reply by apdan510** [reply]
This error is saying that scripts are disabled by a security policy on your pc. Please try this -

Click on the Windows Start button and search for "Powershell" and click on "Powershell - Run as Administrator"
![image](https://user-images.githubusercontent.com/40121410/171687440-4cec1779-d296-442d-a02c-b0df8067ec3c.png)

Enter this command into your Powershell terminal and enter the "A" key to accept: `Set-ExecutionPolicy RemoteSigned`
![image](https://user-images.githubusercontent.com/40121410/171687537-892b18d3-5915-4cdd-880d-9ad796159a38.png)

Try testing Truffle again with: `truffle version`
![image](https://user-images.githubusercontent.com/40121410/171687581-d7c2b367-4c70-44fb-b160-010856f1d492.png)

**Comment by Midsty** [comment]
Thank you so much that fixed the problem!

**Comment by Ahmednayeem8** [comment]
![trerrr](https://user-images.githubusercontent.com/60032408/176658176-e4fa15ca-efc8-415b-81de-dc0178cfb32c.PNG)
followed the steps, tried everything nothing seems to work. on the verge of giving up

**Reply by apdan510** [reply]
@Ahmednayeem8 lets try this.

Install NVM to assist us in downgrading node to version 14
1. https://github.com/coreybutler/nvm-windows/releases/download/1.1.9/nvm-setup.exe

2. After nvm finishes installing, enter this command into your powershell: 

`nvm install 14`
then
`nvm use 14.19.3`

then try installing truffle one more time
`npm install -g truffle`

Let me know if this works

Kind regards,
Alex | Truffle Support

**Reply by Ahmednayeem8** [reply]
Finally after hours of searching your solution got it working for me. THANKS A LOT!

**Reply by Ahmednayeem8** [reply]
wait, its back
![fme](https://user-images.githubusercontent.com/60032408/176906473-711cbaf6-4d92-4f5a-bc67-c7898cb96d50.PNG)

**Reply by apdan510** [reply]
Please try typing truffle version into your terminal and let me know what it returns

**Comment by Rakesh-sahoo07** [comment]
![Screenshot (23)](https://user-images.githubusercontent.com/104811762/178113953-b9754dbd-f8d4-49c0-b29d-a18b571c7b93.png)
please help me

**Reply by Rakesh-sahoo07** [reply]
errors will destroy me

**Reply by apdan510** [reply]
This error is saying that there is no visual studio compiler is installed on your computer, please follow these steps to install it:

**Open an administrator CMD line or Powershell
Make sure Chocolatey is installed by pasting the below command into an administrative Cmdline or Powershell:**
Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))

**Install Visual studio 2019 build tools**
choco install -y visualstudio2019buildtools visualstudio2019-workload-vctools

**Set VS studio 2019 build tools**
npm config set msvs_version 2019

**Then try installing Truffle once more**:
npm install -g truffle

**Comment by schutau** [comment]
Also having issues installing truffle.

node v18.13.0
npm v8.19.3

Install has warnings but no errors, but version check fails:
![image](https://user-images.githubusercontent.com/35646610/212028733-32f3d4cf-cc40-4a2e-af6a-3720762be9b6.png)

Any advice on how to resolve?

**Reply by schutau** [reply]
truffle version leads to an error: "The term 'truffle' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again."

**Comment by Fancy30** [comment]
Kindly check if truffle had been installed by checking the truffle version

**Reply by schutau** [reply]
truffle version leads to an error: "The term 'truffle' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again."

**Reply by schutau** [reply]
Yes, installed node.js, Choclatey, and Microsoft Visual Studio (incl. C++ and CMake). All prereq's should be installed (but not sure if there's version issues that current truffle install doesn't support).

**Reply by schutau** [reply]
Tried that and still same issue. Suspect it might be failing in the first step and not making the directory (based on first few lines of warnings):
---
npm WARN deprecated mkdirp-promise@5.0.1: This package is broken and no longer maintained. 'mkdirp' itself supports promises now, please switch to that.
npm WARN deprecated har-validator@5.1.5: this library is no longer supported

**Comment by Fancy30** [comment]
I solved mine and what i did was reinstall node
then I installed truffle by using npm i and not --g
Also, i ensured I was opening the command prompt as an administrator
You can also check the folder i was trying to install to to be sure of the contents by using dir
Also i then rechecked truffle -v and i unboxed truffle

**Reply by schutau** [reply]
Tried that and still same issue. Suspect it might be failing in the first step and not making the directory (based on first few lines of warnings):
npm WARN deprecated mkdirp-promise@5.0.1: This package is broken and no longer maintained. 'mkdirp' itself supports promises now, please switch to that.
npm WARN deprecated har-validator@5.1.5: this library is no longer supported

**Reply by cipaterson** [reply]
Hi @schutau, those are warnings and unlikely to be related to your problem.  Following the install instructions carefully (https://github.com/orgs/trufflesuite/discussions/5089) should have left you in a good state.  If not we'd be keen to understand what happened - I'll have to do some experiments.

What's the contents of your $PATH?  "The term 'truffle' is not recognized..." suggests that your PATH does not include the path of the newly installed truffle.

Chris

**Reply by schutau** [reply]
Yes, that was it - editing the path and enabling the script execution policy via Set-ExecutionPolicy got things to finally start working. Thanks all!
